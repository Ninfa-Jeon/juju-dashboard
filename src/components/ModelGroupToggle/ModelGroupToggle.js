import React from "react";
import classNames from "classnames";

import "./_model-group-toggle.scss";

const ModelGroupToggle = ({ groupedBy, setGroupedBy }) => {
  return (
    <div className="p-model-group-toggle">
      <div className="p-model-group-toggle__inner">
        <span className="p-model-group-toggle__label">Group by:</span>
        <div className="p-model-group-toggle__buttons">
          <button
            className={classNames("p-model-group-toggle__button", {
              "is-selected": groupedBy === "status"
            })}
            value="status"
            onClick={e => setGroupedBy(e.target.value)}
          >
            Status
          </button>
          <button
            className={classNames("p-model-group-toggle__button", {
              "is-selected": groupedBy === "cloud"
            })}
            value="cloud"
            onClick={e => setGroupedBy(e.target.value)}
          >
            Cloud
          </button>
          <button
            className={classNames("p-model-group-toggle__button", {
              "is-selected": groupedBy === "owner"
            })}
            value="owner"
            onClick={e => setGroupedBy(e.target.value)}
          >
            Owner
          </button>
        </div>
      </div>
    </div>
  );
};

export default ModelGroupToggle;
