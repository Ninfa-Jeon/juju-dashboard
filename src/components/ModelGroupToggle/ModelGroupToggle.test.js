import React from "react";
import { shallow } from "enzyme";

import ModelGroupToggle from "./ModelGroupToggle";

describe("Model group toggle", () => {
  it("renders without crashing and matches snapshot", () => {
    const wrapper = shallow(<ModelGroupToggle />);
    expect(wrapper.find(".p-model-group-toggle")).toMatchSnapshot();
  });

  it("shows active grouping", () => {
    const wrapper = shallow(<ModelGroupToggle groupedBy="cloud" />);
    expect(wrapper.find(".p-model-group-toggle")).toMatchSnapshot();
  });

  it("calls to set group by on click", () => {
    const setGroupedBy = jest.fn();
    const wrapper = shallow(
      <ModelGroupToggle groupedBy="cloud" setGroupedBy={setGroupedBy} />
    );
    expect(
      wrapper.find(".p-model-group-toggle__button.is-selected").text()
    ).toBe("cloud");
    wrapper.find("button[value='owner']").simulate("click", {
      target: { value: "owner" }
    });
    expect(setGroupedBy.mock.calls.length).toBe(1);
    expect(setGroupedBy.mock.calls[0]).toEqual(["owner"]);
    // We don't check that the UI updated because it has no internal state.
    // It requires a parent to re-render it with a new selected group.
  });
});
