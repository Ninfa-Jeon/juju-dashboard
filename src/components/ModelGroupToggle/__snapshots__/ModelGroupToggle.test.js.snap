// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model group toggle renders without crashing and matches snapshot 1`] = `
<div
  className="p-model-group-toggle"
>
  <div
    className="p-model-group-toggle__inner"
  >
    <span
      className="p-model-group-toggle__label"
    >
      Group by:
    </span>
    <div
      className="p-model-group-toggle__buttons"
    >
      <button
        className="p-model-group-toggle__button"
        onClick={[Function]}
        value="status"
      >
        Status
      </button>
      <button
        className="p-model-group-toggle__button"
        onClick={[Function]}
        value="cloud"
      >
        Cloud
      </button>
      <button
        className="p-model-group-toggle__button"
        onClick={[Function]}
        value="owner"
      >
        Owner
      </button>
    </div>
  </div>
</div>
`;
